packages/atomic/cypress/integration/facet-date.cypress.ts

```diff
@@ -0,0 +1,233 @@
+import {setUpPage, shouldRenderErrorComponent} from '../utils/setupComponent';
+import {
+  validateFacetComponentLoaded,
+  validateFacetNumberofValueGreaterThan,
+  validateFacetNumberofValueEqual,
+  assertBasicFacetFunctionality,
+  facetValueShouldDisplayInBreadcrumb,
+  assertNonZeroFacetCount,
+  convertFacetValueToAPIformat,
+  FacetType,
+  DateRange,
+  convertDateToFacetValue,
+  assertDeselectFacet,
+  assertClearAllFacet,
+  convertDateFormatLabel,
+} from './facet-utils';
+
+import {
+  FacetSelectors,
+  createAliasShadow,
+  createAliasFacetUL,
+  createBreadcrumbShadowAlias,
+} from './facet-selectors';
+
+const dateFacetProp = {
+  field: 'date',
+  label: 'Date',
+};
+
+interface DateRanges {
+  range1: DateRange;
+  range2: DateRange;
+  range3: DateRange;
+}
+const dateRange: DateRanges = {
+  range1: {
+    start: '01/01/2006',
+    end: '01/01/2011',
+  },
+  range2: {
+    start: '01/01/2011',
+    end: '01/01/2015',
+  },
+  range3: {
+    start: '01/01/2015',
+    end: '01/01/2021',
+  },
+};
+
+function setupAutoDateFacet(field: string, label: string, option?: string) {
+  setUpPage(
+    `<atomic-breadcrumb-manager></atomic-breadcrumb-manager>    
+    <atomic-date-facet field="${field}" label="${label}" ${option}>
+  </atomic-date-facet>`
+  );
+}
+
+function setupCustomDateFacet(
+  field: string,
+  label: string,
+  dateRange: DateRanges
+) {
+  setUpPage(
+    `<atomic-breadcrumb-manager></atomic-breadcrumb-manager>
+    <atomic-date-facet field="${field}" label="${label}">
+           <atomic-date-range start=${dateRange.range1.start} end=${dateRange.range1.end} end-inclusive="true"></atomic-date-range>
+           <atomic-date-range start=${dateRange.range2.start} end=${dateRange.range2.end} end-inclusive="true"></atomic-date-range>
+           <atomic-date-range start=${dateRange.range3.start} end=${dateRange.range3.end} end-inclusive="true"></atomic-date-range>
```

Comment:
It would be possible to generate the strings using a loop if `dateRanges` is an array.

```
const ranges = dateRanges.map(r => <atomic-date-range start=${r.start} end=${r.end} end-inclusive="true"></atomic-date-range>).join()

const html = `<atomic-date-facet field="${field}" label="${label}">${ranges}</atomic-date-facet>`;
setUpPage(html)
```

Reply:
Sure thanks Sami